### ISSUE TITLE
Build error on Linux (openSUSE Tumbleweed, GCC10)

### ISSUE THREAD

**Issue by guoyunhe** [jpd002/Play-#914]
```
[   65s] /home/abuild/rpmbuild/BUILD/Play-0~git20200602/deps/Framework/build_cmake/Framework/../../include/HashUtils.h:10:54: error: 'size_t' has not been declared
```
After openSUSE Tumbleweed upgrade to GCC10, it failed to build.

**Comment by Zer0xFF** [comment]
> - FreeBSD travis image seems to contain gcc 10, which is causing build problems, mostly due to ["Reduced header dependencies, leading to faster compilation for some code"](https://gcc.gnu.org/gcc-10/changes.html), which we will need to address eventually.

GCC messed up on that one, the issue is more widespread, so i wont be surprised if they end up reversing that.

If anyone is interested in fixing this, its easy, but probably time consuming, based on the approach.

1. using pch to include the missing headers (we already have something[ in place for windows](https://github.com/jpd002/Play-/blob/5fead001e7a46babef38cf563eaa9bbbe1e96bf0/Source/CMakeLists.txt#L456)), might even work without change, cmake 3.17 did also add [target_precompile_headers ](https://cmake.org/cmake/help/v3.17/command/target_precompile_headers.html)
2. add each header as needed
3. downgrade to gcc 9

**Comment by guoyunhe** [comment]
Yes, tons of packages are failing to build. I am totally lost...

**Comment by jpd002** [comment]
Can't we just add an include for `cstddef` to fix that error?

**Comment by guoyunhe** [comment]
I tried but it doesn't work as I expected. Maybe I used it in a wrong way.

**Comment by guoyunhe** [comment]
yes, it is fixed now. thanks!
